alpenhorn_chime/info/base.py

```diff
@@ -0,0 +1,245 @@
+        #
+        # where the datetime has the form: YYYMMDDTHHMMSSZ
+
+        # Split the path on underscores
+        parts = str(path).split("_")
+
+        # If we don't have three parts, the path is invalid
+        if len(parts) != 3:
+            return False
+
+        # Check that the final part is the same as the type name
+        if parts[2] != cls.type().name:
+            return False
+
+        # Check that the second part is a known ArchiveInst
+        try:
+            ArchiveInst.get(name=parts[1])
+        except pw.DoesNotExist:
+            return False
+
+        # Check that the first part is a valid date
+        try:
+            cls.timestamp_from_name(parts[0])
+        except ValueError:
+            return False
+
+        # Otherwise type detection succeeds
+        return True
+
+    @classmethod
+    def timestamp_from_name(cls, path: pathlib.Path | str) -> int:
+        """Return epoch timestamp from acq path `path`
+
+        Parameters
+        ----------
+        path : path.Pathlib or str
+            The name of the acqusition
+
+        Returns
+        -------
+        timestamp : int
+            The seconds-since-epoch of the acq time
+            encoded in the acq path.
+
+        Raises
+        ------
+        ValueError
+            path was not a valid date
+        """
+        d = datetime.datetime.strptime(str(path)[0:16], "%Y%m%dT%H%M%SZ")
+        return calendar.timegm(d.utctimetuple())
+
+
+class CHIMEAcqInfo(CHIMEAcqDetect, acq_info_base):
+    """Abstract base class for CHIME Acq Info tables.
+
+    In addition to the type detection provided by CHIMEAcqDetect,
+    this adds the standard Alpenhorn AcqInfo base table.
+
+    Provides an implementation of `_set_info` for convenience which
+    converts the call into a call to `_info_from_file`.
+
+    Subclasses should add additional fields.  They should also
+    either reimplement `_set_info` or else implement the method
+    `_info_from_file` which will be passed an open, read-only
+    file object.
+    """
+
+    def _set_info(
+        self, node: StorageNode, path: pathlib.Path, item: ArchiveAcq
+    ) -> dict:
+        """Set info for new ArchiveAcq `item` from file `path`.
+
+        Converts the standard Alpenhorn _set_info call
+        into a `_info_from_file` call, if that method exists.
+
+        Parameters
+        ----------
+        node : StorageNode
+            the node we're importing on
+        path : pathlib.Path
+            the path relative to `node.root` of the file
+            being imported in this acquistion
+        item : alpenhorn.archive.ArchiveAcq
+            the newly-created acq record
+
+        Returns
+        -------
+        info : dict
+            any data returned by `_info_from_file`, if that
+            method exists, or else an empty dict.
+        """
+        if hasattr(self, "_info_from_file"):
+            # Open the file
+            file = node.io.open(path)
```

Comment:
I can't find an implementation of the Node IO with an `open` method. Is that already in a PR?

Regardless, this should be implemented and utilised as a context manager.

Reply:
Yeah, `io.open` is somewhere in the PR stack for alpenhorn.  It returns a regular file object, as if `open()` were called, so it's entirely possible to do:
```suggestion
            with node.io.open(path) as file:
```

but is that going to create trouble with h5py?  The code then becomes, effectively:
```
with node.io.open(path) as file:
    with h5py.File(file, "r") as f:
        # Do stuff with HDF5 file "f"
```
Does `h5py.File` close the underlying `file` when its context exits?  If it does, is that a problem for the outer context which will presumably also try to close the file?